iceoryx_posh/include/iceoryx_posh/runtime/posh_runtime.hpp

```diff
@@ -146,8 +146,10 @@ class PoshRuntime
     ///        currently each request is followed by a response
     /// @param[in] msg request message to send
     /// @param[out] response from the RouDi daemon
+    /// @param[in] timeout receive timeout, default disabled
     /// @return true if sucessful request/response, false on error
-    virtual bool sendRequestToRouDi(const IpcMessage& msg, IpcMessage& answer) noexcept = 0;
+    virtual bool
+    sendRequestToRouDi(const IpcMessage& msg, IpcMessage& answer, iox::units::Duration timeout = 0_s) noexcept = 0;
```

Comment:
Please use `iox::Optional<iox::units::Duration> timour = iox::nullopt` instead of a surrogate value.